lib/nerves_runtime.ex

```diff
@@ -68,6 +68,21 @@ defmodule Nerves.Runtime do
     end
   end
 
+  @doc """
+  Uses `boardid` to return a unique serial number.
+  """
+  @spec serial_number() :: String.t() | no_return()
+  def serial_number() do
+    case System.find_executable("boardid") do
+      nil ->
+        raise("boardid was not found. The Nerves System must be fixed to include it!")
```

Comment:
I'm worried that raising will be hard to debug since this could be called in critical code. How about changing this to return a default and documenting the default. I can't think of anything great - perhaps `"unconfigured"`?